### DISCUSSION TITLE
Writing values to the Tesla API via MQTT

### DISCUSSION THREAD

**Discussion by shadow7412** [teslamate-org/teslamate#1083] [Q&A]
I'm using Home Assistant - and after realising that the default Tesla integration is hungry as a hippo, I've decided to go with TeslaMate's MQTT approach. For reading data, this works really well. But losing the ability to trigger things (such as climate control) is a step backwards.

[This information ](https://docs.teslamate.org/docs/integrations/home_assistant)in the documentation intrigued me;

> If however you would like to be able to write values to the Tesla API (Lock/Unlock Doors or automate Climate), there is a solution which involves configuring an extremely high polling interval for the Tesla component and using automation to populate the values from the TeslaMate MQTT parameters.

But as far as I could find, there's no elaboration on how one actually does this. Can we have this added to the docs?

**Comment by brendanpward** [comment]
I understood this to mean that we read from TeslaMate, but write to the Home Assistant integration.  For example, I could query TeslaMate to see if the doors are locked after midnight, but use HA Tesla to send the lock command.  It just means you have to be logged in twice.

**Comment by shadow7412** [comment]
Hmm... I didn't notice any configuration implying that we could stop the HA Tesla spam keeping the vehicle awake - which was the whole point of wiring up the MQTT channels in the first place.

If you're logged in via the HA integration, then there's no point wiring up the TeslaMate one as the data is already there, right?

**Comment by brendanpward** [comment]
I think this is the reason for "configuring an extremely high polling interval", ie. HA Tesla polls once each day (set scan_interval: 86400000).  TeslaMate uses its more advanced polling interval to query the state of the vehicle which won't wake it up as frequently as HA would if its polling interval were set lower.

In other words, HA Tesla will receive accurate data once per day during its single query.  So don't rely on any of the data received from that integration.  Rely on the data from TeslaMate instead as it's updated more frequently through the day.

In other other words, HA Tesla is only to be written to and TeslaMate is only to be read from.

**Comment by shadow7412** [comment]
Oooh I see. Yeah, ok that makes sense.
